compiler/resolution/functionResolution.cpp

```diff
@@ -8883,3 +8899,25 @@ setScalarPromotionType(AggregateType* ct) {
       ct->scalarPromotionType = field->type;
   }
 }
+
+//
+// Does this function call a ref-return function that relies on the
+// setter param?  By induction this handles the general recursive case.
+//
+static bool functionUsesSetter(FnSymbol* fn) {
+  std::vector<CallExpr*> calls;
+  bool                   retval = false;
+
+  collectCallExprs(fn, calls);
+
+  for (size_t i = 0; i < calls.size() && retval == false; i++) {
```

Comment:
Would a "for_vector" macro with a "break" statement be better here? I think this could go in as-is, but doesn't match the usual compiler conventions.